sourcemash/frontend/static/js/views/help_view.js

```diff
@@ -0,0 +1,43 @@
+Sourcemash.Views.HelpView = Backbone.View.extend({
+    template: JST['help'],
+
+    initialize: function(options) {
+        this.user = options.user;
+    },
+
+    events: {
+        'click .help-fixed-action-btn': 'showHelpModal',
+        'click #go-to-profile': 'showProfileOrRegisterModal',
+        'click #focus-login': 'activateLoginForm',
+        'click #show-register-modal': 'showRegisterModal',
+    },
+
+    showHelpModal: function() {
+        $("#help-modal").openModal();
+    },
+
+    showProfileOrRegisterModal: function(e) {
+        e.preventDefault();
+        $("#help-modal").closeModal();
+        if (this.user.get('email')) {
+            window.location.href = "/#profile";
+        } else {
+            setTimeout(function(){$("#register-modal").openModal();}, 250);
```

Comment:
When you play with it, it's clear that you need a small delay (250 milliseconds) between the first modal closing and the second one opening. Otherwise, it's just an explosion of the screen going white (modal closed) and then dark (new modal open) that's hard to look at.